File: src/main/java/reading/excel/ExcelReader.java
package reading.excel;

import jxl.Sheet;
import jxl.Workbook;
import jxl.read.biff.BiffException;

import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ExcelReader {

    public static Map<Integer, List<String>> readJExcel(String fileLocation)
            throws IOException, BiffException {

        Map<Integer, List<String>> data = new HashMap<>();

        Workbook workbook = Workbook.getWorkbook(new File(fileLocation));
        Sheet sheet = workbook.getSheet(0);
        int rows = sheet.getRows();
        int columns = sheet.getColumns();

        for (int i = 0; i < rows; i++) {
            data.put(i, new ArrayList<String>());
            for (int j = 0; j < columns; j++) {
                data.get(i)
                        .add(sheet.getCell(j, i)
                                .getContents());
            }
        }
        return data;
    }
}

File: src/main/java/reading/excel/ExcelReader.kt
package reading.excel

import jxl.Workbook
import jxl.read.biff.BiffException
import java.io.File
import java.io.IOException

object ExcelReader {
    @Throws(IOException::class, BiffException::class)
    fun readJExcel(fileLocation: String): Map<Int, MutableList<String>> {
        val data: MutableMap<Int, MutableList<String>> = HashMap()
        val workbook = Workbook.getWorkbook(File(fileLocation))
        val sheet = workbook.getSheet(0)
        val rows = sheet.rows
        val columns = sheet.columns
        for (i in 0 until rows) {
            data[i] = ArrayList()
            for (j in 0 until columns) {
                data[i]
                    ?.add(
                        sheet.getCell(j, i)
                            .contents
                    )
            }
        }
        return data
    }
}
